config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/gateway/ConnectionUtils.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2018. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.util.gateway;
+
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.builder.EntityBuilderException;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+
+public class ConnectionUtils {
+
+    /**
+    * SSLSocketFactory used to retrieve certificates even if they are not in the trust store,
+     * to avoid SSL handshake exceptions upon socket connection. This socket factory should only be used to read cert
+     * information, and should not be used for persistent TLS connections.
+    * @return the sslSocketFactory that accepts all connections.
+     */
+    public static SSLSocketFactory createAcceptAllSocketFactory() {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
+            sslContext.init(null,
+                    new TrustManager[]{new X509TrustManager() {
+                        @Override
+                        public void checkClientTrusted(X509Certificate[] x509Certificates, String s) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. ([squid:S1186](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1186))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=68&issues=AWX42AqEOlZ5QoPzNuUx&open=AWX42AqEOlZ5QoPzNuUx)

[ISSUE_KEY]: AWX42AqEOlZ5QoPzNuUx